### DISCUSSION TITLE
Array equations with delay generate errors

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#10970] [Q&A]
Hi, 

Defining:
```
Real ys[Nso];
Real ysw[Nso];
```
with a integer dimension, the vectorial equation (tau is parameter Real):
`ysw = ys - delay(ys, tau);`
gives compiling errors, that may be solved by:
`ysw = ys - delay(ys, tau, tau);`

I think that the third argument should not be needed. 

However, the simulation generates zero crossing events each 6 seconds. In the case of Nso = 3, this is the output  each 6 seconds (AUC is the component):
[7] delayZeroCrossing(0, 6, AUC.ys[1], 3600.0, 3600.0) > 0.0
[6] delayZeroCrossing(0, 5, AUC.ys[2], 3600.0, 3600.0) > 0.0
[5] delayZeroCrossing(0, 4, AUC.ys[3], 3600.0, 3600.0) > 0.0

This seems an error related to the issue #8150. Converting the vectorial equation in a loop of scalar equations solves the problem:
```
for i in 1:Nso loop
  ysw[i] = ys[i] - delay(ys[i], tau);
end for;
```

I think that it is also relevant that if arrays ys and ysw has enumerated dimensions, then the equation 
`ysw = ys - delay(ys, tau);`
is not solved by 
`ysw = ys - delay(ys, tau, tau);`
as if delay vectorial equations can only manage integer dimensions.

System: latest nightly build OM 1.22 dev on Windows 11.

Thank you for your support!

**Comment by casella** [comment]
This is related to #8565. Unfortunately I'm afraid we don't have the resource to handle this right now.
